l1-contracts/contracts/chain-registrar/ChainRegistrar.sol

```diff
@@ -0,0 +1,180 @@
+            baseToken: BaseToken({
+                tokenAddress: tokenAddress,
+                tokenMultiplierSetter: tokenMultiplierSetter,
+                gasPriceMultiplierNominator: gasPriceMultiplierNominator,
+                gasPriceMultiplierDenominator: gasPriceMultiplierDenominator
+            })
+        });
+        bytes32 key = keccak256(abi.encode(msg.sender, config.chainId));
+        if (deployedChains[key] || bridgehub.stateTransitionManager(config.chainId) != address(0)) {
+            revert ChainIsAlreadyDeployed();
+        }
+        proposedChains[key] = config;
+        // For Deploying L2 contracts on for non ETH based networks, we as bridgehub owners required base token.
+        if (config.baseToken.tokenAddress != ETH_TOKEN_ADDRESS) {
+            uint256 amount = (1 ether * config.baseToken.gasPriceMultiplierNominator) /
+                config.baseToken.gasPriceMultiplierDenominator;
+            if (IERC20(config.baseToken.tokenAddress).balanceOf(l2Deployer) < amount) {
+                bool success = IERC20(config.baseToken.tokenAddress).transferFrom(msg.sender, l2Deployer, amount);
+                if (!success) {
+                    revert BaseTokenTransferFailed();
+                }
+            }
+        }
+        emit NewChainRegistrationProposal(config.chainId, msg.sender, key);
+    }
+
+    // @dev Change l2 deployer
+    function changeDeployer(address newDeployer) public onlyOwner {
+        l2Deployer = newDeployer;
+        emit L2DeployerChanged(l2Deployer);
+    }
+
+    function getChainConfig(address author, uint256 chainId) public view returns (ChainConfig memory) {
+        bytes32 key = keccak256(abi.encode(author, chainId));
+        return proposedChains[key];
+    }
+
+    // @dev Get data about the chain that has been fully deployed
+    function getRegisteredChainConfig(uint256 chainId) public view returns (RegisteredChainConfig memory) {
+        address stm = bridgehub.stateTransitionManager(chainId);
+        if (stm == address(0)) {
+            revert ChainIsNotYetDeployed();
+        }
+        address diamondProxy = IStateTransitionManager(stm).getHyperchain(chainId);
+
+        address pendingChainAdmin = IGetters(diamondProxy).getPendingAdmin();
+        address chainAdmin = IGetters(diamondProxy).getAdmin();
+        address l2BridgeAddress = bridgehub.sharedBridge().l2BridgeAddress(chainId);
+        if (l2BridgeAddress == address(0)) {
+            revert BridgeIsNotRegistered();
+        }
+
+        RegisteredChainConfig memory config = RegisteredChainConfig({
+            pendingChainAdmin: pendingChainAdmin,
+            chainAdmin: chainAdmin,
+            diamondProxy: diamondProxy,
+            l2BridgeAddress: l2BridgeAddress
+        });
+        return config;
+    }
+
+    // @dev Mark chain as registered. Emit necessary events for spinning up the chain server
+    function setChainAsRegistered(address author, uint256 chainId) public onlyOwner nonReentrant {
```

Comment:
Why you need to set it is as registered?

Reply:
for spawning necessary events, and marking it as deployed. 
And the most important part is that when we move to the step2, we will have similar setup. with exact same events

Reply:
decided to give up on this function and implement properly in the next release